pkg/traceability/apiExceptionList.go

```diff
@@ -0,0 +1,58 @@
+package traceability
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/Axway/agent-sdk/pkg/util/log"
+)
+
+func setUpAPIExceptionList(apiExceptionsList []string) (string, error) {
+	// if set check for valid regex definitions
+	if apiExceptionsList != nil {
+		newShowRegex = make([]*regexp.Regexp, 0)
+
+		// Get the api exceptions list
+		exceptions := apiExceptionsList
+		for i := range exceptions {
+			exception := strings.TrimSpace(exceptions[i])
+
+			// check for regex and then validate
+			keyMatch, err := regexp.Compile(exception)
+			if err != nil {
+				return exception, err
+			}
+
+			newShowRegex = append(newShowRegex, keyMatch)
+
+		}
+	}
+	return "", nil
+}
+
+// getAPIExceptionsList - Returns traceability APIs exception list (api paths)
+func getAPIExceptionsList() []*regexp.Regexp {
+	if newShowRegex == nil {
+		return []*regexp.Regexp{}
+	}
+	return newShowRegex
+}
+
+// newShowRegex - array of regexp.Regexp
+var newShowRegex []*regexp.Regexp
```

Comment:
Can we wrap this variable in an struct/interface for managing the exception list and then setup can construct it? Something like
type APIExceptionList interface {
    ShouldIgnoreEvent(uriRaw string) bool // or some better method name 
}

type apiExceptionList struct {
    exceptionRegEx []*regexp.Regexp
}

func (al *apiExceptionList) ShouldIgnoreEvent(uriRaw string) bool {
  // use al.exceptionReg list to perform check
   ...
}

Then setupAPIExceptionList() can construct the apiExceptionList instance based on the config.

Don't like keeping global variable, but I understand for this case we will need to keep global variable for APIExceptionList interface as the object is constructed while parsing the traceability config.

Reply:
Nevermind, we will anyway have to keep a global variable and then will have to add a method to return the constructed interface. Would have been good if we could store the constructed interface somewhere that can be referenced by other packages.

So, please ignore my comment

Reply:
@vivekschauhan no worries.. Still updated to pear down unneeded code.  Re-check when you can... thank you.